### ISSUE TITLE
Auto-detection of baseline ploidy

### ISSUE THREAD

**Issue by vladimirg** [berman-lab/ymap#52]
Today, in order to correctly assign the allelic ratios (and assign colors in the SNP diagrams created with a hapmap) the user needs to provide the correct baseline ploidy. This tells Ymap which possible ratios the SNPs can fall into. For example, in a diploid segment, the only possible SNP ratios are 2:0, 1:1, and 0:2. However, if the segment is triploid, then the actual ratios are 3:0, 2:1, 1:2 and 0:3. Only the homozygotes can be correctly assigned a color, and the rest will be either strictly incorrect or appeared smeared.

Many times the user can only guess the baseline ploidy, and sometimes they even incorrectly assume it, and don't look at the allele ratio diagrams to re-run the dataset. So it may be a good idea to introduce an automatic detection of the baseline ploidy.

The working suggestion is this: take all segments with CNV matching the baseline ploidy. Find the peaks of the allele ratios within them. Match the ratios those peaks represent with a ploidy, and use that as the baseline ploidy in the rest of the analysis. So, for example, if we see a peak at 0 and a peak at 0.75, we can assume that the baseline ploidy is 4 (this was actually observed in a strain assumed to be haploid).

@darrenabbey , what do you think?

**Comment by darrenabbey** [comment]
Some terminology may need to be clarified. "Baseline Ploidy" as originally
written is used in figure generation only, while "Experimental Ploidy" (I
forget exact use. I'll have to check later) is used to scale normalized CNV
data and thus determine theoretical SNP ratio peaks.

**Comment by vladimirg** [comment]
Ah, you're right! But it doesn't change the proposal, and we'd love to hear your thoughts about it.

**Comment by darrenabbey** [comment]
Yeah. I didn't want to go ahead until I was sure we were speaking the same
language. I'll get you a more full response once I get home.

**Comment by darrenabbey** [comment]
I don't like the idea of automatically over-riding the user inputed
experimental ploidy. However, such an analysis could be very helpful in
that the results could be highlighted in the user interface as possibly
requiring a re-analysis due to evidence for a different overall ploidy than
what was indicated.

**Comment by vladimirg** [comment]
We were thinking about having a combo box with two options: "auto-detect" and "manual". In manual mode the user can enter the ploidies as before, and the code will run as before. In "auto-detect" mode, the experimental ploidy is auto-detected and also used as the baseline for the figure.

If the auto-detection works well, it seems to me that it should be the default. So the question is, can our proposed strategy be expected to work?
